packages/react-scripts/scripts/build.js

```diff
@@ -57,6 +58,34 @@ measureFileSizesBeforeBuild(paths.appBuild)
     fs.emptyDirSync(paths.appBuild);
     // Merge with the public folder
     copyPublicFolder();
+
+    // Check if every vendors are defined in the package.json
+    // @remove-on-eject-begin
+    // The devDepencencies shouldn't be available for vendors
+    // But otherwise the tests fail.
+    // @remove-on-eject-end
+    const packageJson = require(paths.appPackageJson);
+    const dependencies = Object.keys(packageJson.dependencies || {})
+      .concat(Object.keys(packageJson.devDependencies || {}))
+      .concat(Object.keys(packageJson.peerDependencies || {}));
+    const vendors = extractVendors();
+    const flatVendors = [];
+    Object.keys(vendors).forEach(vendorName => {
+      flatVendors.concat(vendors[vendorName]);
+    });
+
+    const missingVendors = Object.keys(flatVendors).filter(
+      vendor => dependencies.indexOf(vendor) === -1
+    );
```

Comment:
`flatVendors` is meant to be an array, correct? If yes, why won't we just filter over itself?
```js
const missingVendors = flatVendors.filter(
  vendor => dependencies.indexOf(vendor) === -1
);
```